### PR TITLE
ES6: common/modules/navigation/navigation.js

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Convert common/modules/navigation/navigation.js to ES6.

## What is the value of this and can you measure success?

More ES6!

## Does this affect other platforms - Amp, Apps, etc?

No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

No

## Screenshots

N/A

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/commercial/donot-use-adblock.js [hunk 1]
```diff
@@ -10,7 +10,6 @@ define([
     'common/modules/adblock-banner',
     'common/modules/onward/history',
     'common/modules/ui/message',
-    'common/modules/navigation/navigation',
```
Comment (GHaberis) [inline static/src/javascripts-legacy/projects/common/modules/commercial/donot-use-adblock.js]: `common/modules/navigation/navigation.js` was imported and never used by this module

**Review by gu-stav — APPROVED** [review]

**Comment by GHaberis** [comment]
@gustavpursche I've had to refactor a little to make this module testable. `initNavigation` now returns a `Promise.all` and each function that modifies the dom now returns a `fastdom` promise. This means I can chain calls to `initNavigation` with a `then` in the tests. Would you be able to give it a quick :eyes:. Thanks!

File: static/src/javascripts/projects/common/modules/navigation/navigation.spec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    </div>
+    <div class="${NAVIGATION_CONTAINER_CLASS}"></div>`;
+
+    beforeEach(() => {
+        if (document.body) {
+            document.body.innerHTML = domSnippet;
+        }
+    });
+
+    afterEach(() => {
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/navigation/navigation.spec.js]: Is this necessary? You reset the DOM beforeEach aswell ...

**Review by gu-stav — APPROVED** [review]
SPECS 😍

File: static/src/javascripts/projects/common/modules/navigation/navigation.spec.js (hunk 2 above)
Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/navigation/navigation.spec.js]: True, I guess from working with `mocha` / `jasmine` / `phantom` in the past I'd always do this to ensure that after the final test had run the DOM was cleaned for subsequent specs. I wondered whether it was necessary with `Jest` so looked elsewhere in our `Jest` specs and this has been done.

**Pull request merged**